### ISSUE TITLE
Improve documentation

### ISSUE THREAD

**Issue by rollcat** [rollcat/judo#12]
The original readme was written in 2016 and reflected my frustration with Ansible. It tries to explain the motivation/design philosophy, provide a full reference, and serve as a tutorial with examples - all at once, and as an effect it doesn't do any of the three too well.

Perhaps it can be split up for these three purposes: the readme can still serve as an introduction; an actual (installable!) man page would be good as a reference; and the examples could be elaborated on, with some actual real-world scripts I've accumulated over the past 6 years.

**Comment by haary** [comment]
I just wanted to open an issue, but now I think it will suffice to comment here. I was getting timeout errors and found out that I could just use judo -t <time> to define the timeout span. The problem was that -t is not documented and I had to study the source code to find out. So, it would be nice to document all parameters (haven't found out yet what  -f and -e KEY does) :-)

**Comment by rollcat** [comment]
@haary agreed, and then there's `-f` which I've wanted to control parallelism (aka forkiness), which is mentioned in `-h`... and is a no-op ;) definitely deserves a clean-up
